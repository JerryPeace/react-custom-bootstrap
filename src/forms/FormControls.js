import React from 'react';
import classNames from 'classnames';
import bootstrapUtils from '../utils/bootstrapUtils';

class FormControls extends React.Component {
    static defaultProps = {
        bsClass: 'controls'
    };

    render() {
        const props = this.props;
        let cls = bootstrapUtils.getCls(props);

        return (
            <div className={classNames(props.className, cls)}>
                {props.children}
            </div>
        )
    }
}

export default FormControls;

