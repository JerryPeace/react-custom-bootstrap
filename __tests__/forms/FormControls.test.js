import React from 'react';
import { mount } from 'enzyme';
import toJSON from'enzyme-to-json';
import FormControls from 'forms/FormControls';

describe('FormControls', () => {
    test('render form controls correctly', () => {
        const form = mount(
            <FormControls>
                <div>test</div>
            </FormControls>
        )
        expect(toJSON(form)).toMatchSnapshot();
    });
});
