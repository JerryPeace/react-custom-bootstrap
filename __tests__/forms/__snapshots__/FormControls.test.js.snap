// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControls render form controls correctly 1`] = `
<FormControls
  bsClass="controls"
>
  <div
    className="controls"
  >
    <div>
      test
    </div>
  </div>
</FormControls>
`;
